src/main/java/com/sap/cloudfoundry/client/facade/domain/CloudRoute.java

```diff
@@ -25,14 +24,35 @@ public int getAppsUsingRoute() {
     @Nullable
     public abstract String getPath();
 
-    @Override
-    public String getName() {
-        return getHost() + "." + getDomain();
-    }
+    @Nullable
+    public abstract Integer getPort();
+
+    public abstract String getUrl();
 
     @Override
     public CloudRoute derive() {
         return this;
     }
 
+    @Override
+    public String toString() {
+        return getUrl();
+    }
+
+    @Override
+    public int hashCode() {
+        return getUrl().hashCode();
+    }
+
+    @Override
+    public boolean equals(Object object) {
+        if (object == this) {
+            return true;
+        }
+        if (!(object instanceof CloudRoute)) {
+            return false;
+        }
+        return getUrl().equals(((CloudRoute) object).getUrl());
```

Comment:
Consider this:
host: test.test
domain: domain
=> url is => test.test.domain
And:
host: test
domain: test.domain
=> url is => test.test.domain

This is NOT the same route

Reply:
Should the hash code calculation be changed too?